core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -481,6 +493,11 @@ class RequestChannel(val queueSize: Int,
 
   def sendShutdownRequest(): Unit = requestQueue.put(ShutdownRequest)
 
+  def sendCallbackRequest(request: CallbackRequest): Unit = {
+    callbackQueue.put(request)
+    requestQueue.put(RequestChannel.WakeupRequest)
```

Comment:
This should be .offer -- we don't need to block if the request queue is full, and it's ok if we don't have a wakeup request in a full queue -- the queue would would contain a request (due to the fact that it's full) to wake up the poll.

Reply:
Offer was throwing spotbugs errors since I didn't check the response. I can do that but it will be a little uglier.

Reply:
I guess, if the request queue is full then it means that the request threads are all overloaded, so if we block the inter-broker channel thread it probably wouldn't matter much as the broker is already in pain.  But on the other hand, it's kind of strange to not implement technically more correct behavior because we've got some spurious warnings.

Reply:
The build will fail with the warnings. But I can add the extra code so the build passes.